src/hooks/useSystemColorScheme.ts

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import type { ColorScheme } from "../ColorContext";
+
+function darkModeMatcher(): MediaQueryList | undefined {
+  const media = window?.matchMedia?.("(prefers-color-scheme: dark)");
+  if (!media) {
+    return undefined;
+  }
+
+  return media;
+}
+
+function getSchemeFromMatcher(matcher: MediaQueryList): ColorScheme {
+  return matcher.matches ? "dark" : "light";
+}
+
+function getSchemeFromEvent(event: MediaQueryListEvent): ColorScheme {
+  return event.matches ? "dark" : "light";
+}
```

Comment:
This can be unified by using a union type:

```suggestion
function getScheme(event: MediaQueryList | MediaQueryListEvent): ColorScheme {
  return event.matches ? "dark" : "light";
}
```